lib/bolt/catalog.rb

```diff
@@ -65,6 +65,9 @@ def compile_catalog(request)
                           bolt_inventory: setup_inventory(request['inventory'])) do
             Puppet.lookup(:pal_current_node).trusted_data = target['trusted']
             pal.with_catalog_compiler do |compiler|
+              # Override undesirable settings in the CatalogCompiler. We want Bolt to be able
+              # to compile most Puppet 4+ manifests, so we need to allow deprecated functions.
+              Puppet[:strict] = :warning
```

Comment:
Will this revert back to `:error` at the end of this block?

Reply:
I don't think so. We use `PAL#with_catalog_compiler`, which calls https://github.com/puppetlabs/puppet/blob/master/lib/puppet/parser/catalog_compiler.rb#L15-L16 first. I don't think it resets afterwards.